data_model/derive/src/api.rs

```diff
@@ -0,0 +1,168 @@
+use proc_macro2::TokenStream;
+use quote::quote;
+use syn::Attribute;
+
+fn expose_ffi(mut attrs: Vec<Attribute>, item: &TokenStream) -> TokenStream {
+    let mut ffi_attrs = attrs.iter().filter(|&attr| attr.path.is_ident("ffi_type"));
+
+    if ffi_attrs.next().is_none() {
+        return quote! {
+            #(#attrs)*
+            #item
+        };
+    }
+
+    attrs.retain(|attr| *attr != syn::parse_quote! (#[ffi_type]));
+    let no_ffi_attrs: Vec<_> = attrs
```

Comment:
I don't quite get what this does. What's the difference between `!attr.path.is_ident("ffi_type")` and `attr != syn::parse_quote! (#[ffi_type])`?

Reply:
Got it, thanks